crates/pera-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -122,18 +141,54 @@ impl DWalletMPCManager {
         authority_name: AuthorityName,
         session_id: ObjectID,
     ) -> PeraResult<()> {
+        if self.malicious_actors.contains(&authority_name) {
+            return Ok(());
+        }
         let Some(instance) = self.mpc_instances.get_mut(&session_id) else {
-            error!(
+            warn!(
                 "received a message for instance {:?} which does not exist",
                 session_id
             );
-            // TODO (#261): Punish a validator that sends a message related to a non-existing mpc instance
+            self.malicious_actors.insert(authority_name);
             return Ok(());
         };
-        instance.handle_message(DWalletMPCMessage {
+        let handle_message_response = instance.handle_message(DWalletMPCMessage {
             message: message.to_vec(),
             authority: authority_name,
-        })
+        });
+        if let Err(PeraError::DWalletMPCMaliciousParties(malicious_parties)) =
+            handle_message_response
+        {
+            self.store_malicious_parties_by_indices(malicious_parties)?;
+            return Ok(());
+        };
+        handle_message_response
+    }
+
+    /// Convert the indices of the malicious parties to their addresses and store them
+    /// in the malicious actors set
+    /// New messages from these parties will be ignored
+    fn store_malicious_parties_by_indices(
+        &mut self,
+        malicious_parties: Vec<PartyID>,
+    ) -> PeraResult {
+        let malicious_parties_names = malicious_parties
+            .into_iter()
+            .map(|party_id| {
+                Ok(self
+                    .epoch_store()?
+                    .committee()
+                    .authority_by_index(party_id as u32)
+                    .ok_or(PeraError::InvalidCommittee("".to_string()))?
+                    .clone())
+            })
+            .collect::<PeraResult<Vec<AuthorityName>>>()?;
+        warn!(
+            "found malicious parties while advancing, {:?}",
+            malicious_parties_names
+        );
```

Comment:
This is presumptuous. Who is to say this function was called during "advancing"? Consider 
```suggestion
        warn!(
            "flagged the following parties as malicious: {:?}",
            malicious_parties_names
        );
```